### ISSUE TITLE
Some node names cause errors+wrong results

### ISSUE THREAD

**Issue by andrew-healey** [jriecken/dependency-graph#46]
## Underlying problem

Since `DepGraph` uses an `Object`, and not a `Map`, to store `nodes`, `outgoingEdges`, and `incomingEdges` , it mishandles nodes with specific names.

Two examples of how `Object`s can cause bugs:

```js
Object.keys({__proto__:"hey"}) // []
```
```js
const obj = {};
if(obj["constructor"]) console.log("key 'constructor' is in obj") // key 'constructor' is in obj
```

## Concrete bugs

Two corresponding examples of bugs in `dependency-graph`:

```js
var DepGraph = require('dependency-graph').DepGraph;
 
var graph = new DepGraph();
graph.addNode("a");
graph.addNode('__proto__');
graph.addDependency("a","__proto__") // error: node does not exist: __proto__
```
```js
var DepGraph = require('dependency-graph').DepGraph;
var graph = new DepGraph();
graph.addNode("a");
graph.addNode("constructor");
graph.overallOrder() // ["a"] (notice "constructor" is missing)]
```

## Fix

These bugs can be solved by replacing `this.nodes`, `this.incomingEdges`, and `this.outgoingEdges` with `Map`s.

For example:
`this.nodes = {};` maps to `this.nodes = new Map`,
`this.nodes.hasOwnProperty(node)` maps to `this.nodes.has(node)`.
`this.nodes[node] = node` maps to `this.nodes.set(node,node)`.

**Comment by jriecken** [comment]
Thanks, I've been meaning to do this for a while - I'll need to release a 1.0 version as the current version works with ancient versions of JavaScript that don't have `Map` and so this will be a breaking change. I'll combine it with a few other minor changes as well.

**Comment by jriecken** [comment]
I'm going to switch to use Map/Set - I'd rather not use a hack.

See c9238dceba22188f1c74d025af37fea20744bc03

**Comment by jriecken** [comment]
I've published version `1.0.0` with this change (and your other small fix!)

**Comment by andrew-healey** [comment]
Nice! Congrats on 1.0.0!
